fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFeature.kt

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.selector.findTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.translate.initialFromLanguage
+import mozilla.components.concept.engine.translate.initialToLanguage
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import org.mozilla.fenix.R
+
+/**
+ * Helper for observing Translation state from browserStore.
+ */
+class TranslationsDialogFeature(
+    val browserStore: BrowserStore,
+    private val translationsDialogStore: TranslationsDialogStore,
+    private val sessionId: String,
+    private val context: Context,
+) : LifecycleAwareFeature {
```

Comment:
I would suggest implementing `AbstractBinding` instead of `LifecycleAwareFeature` because it does a bit more work and is more directly intended to bind a view to observations of a Store

Reply:
Added AbstractBinding and unittests.